dev/wddr/device.c

```diff
@@ -528,404 +537,246 @@ void wddr_enable_loopback(wddr_dev_t *wddr)
-
-        // Set Receiver Register in one write
-        reg_val = UPDATE_REG_FIELD(DDR_DQ_DQS_RX_IO_CMN_M0_R0_CFG__POR,
-                                   DDR_DQ_DQS_RX_IO_CMN_M0_R0_CFG_CAL_P_C,
-                                   common_cal->rank[rank_index].dqs.receiver.cal_code[REC_P_SIDE][REC_C_SIDE]);
+        // REN PI
+        reg_val = UPDATE_REG_FIELD(0x0, DDR_DQ_DQS_RX_REN_PI_M1_R0_CFG_CODE, cfg->rank[rank].dqs.pi.ren.code);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_REN_PI_M1_R0_CFG_GEAR, cfg->rank[rank].dqs.pi.ren.gear);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_REN_PI_M1_R0_CFG_XCPL, cfg->rank[rank].dqs.pi.ren.xcpl);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_REN_PI_M1_R0_CFG_EN, 0x1);
+        dq_reg->DDR_DQ_DQS_RX_REN_PI_CFG[msr][rank] = reg_val;
+
+        // RCS PI
+        reg_val = UPDATE_REG_FIELD(0x0, DDR_DQ_DQS_RX_RCS_PI_M1_R0_CFG_CODE, cfg->rank[rank].dqs.pi.rcs.code);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RCS_PI_M1_R0_CFG_GEAR, cfg->rank[rank].dqs.pi.rcs.gear);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RCS_PI_M1_R0_CFG_XCPL, cfg->rank[rank].dqs.pi.rcs.xcpl);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RCS_PI_M1_R0_CFG_EN, 0x1);
+        dq_reg->DDR_DQ_DQS_RX_RCS_PI_CFG[msr][rank] = reg_val;
+
+        // RDQS PI
+        reg_val = UPDATE_REG_FIELD(0x0, DDR_DQ_DQS_RX_RDQS_PI_0_M1_R0_CFG_CODE, cfg->rank[rank].dqs.pi.rdqs.code);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RDQS_PI_0_M1_R0_CFG_GEAR, cfg->rank[rank].dqs.pi.rdqs.gear);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RDQS_PI_0_M1_R0_CFG_XCPL, cfg->rank[rank].dqs.pi.rdqs.xcpl);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_RDQS_PI_0_M1_R0_CFG_EN, 0x1);
+        dq_reg->DDR_DQ_DQS_RX_RDQS_PI_0_CFG[msr][rank] = reg_val;
+
+        // SDR LPDE
+        reg_val = UPDATE_REG_FIELD(0x0, DDR_DQ_DQS_RX_SDR_LPDE_M1_R0_CFG_GEAR, cfg->rank[rank].dqs.sdr_lpde.gear);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_SDR_LPDE_M1_R0_CFG_CTRL_BIN, cfg->rank[rank].dqs.sdr_lpde.delay);
+        reg_val = UPDATE_REG_FIELD(reg_val, DDR_DQ_DQS_RX_SDR_LPDE_M1_R0_CFG_EN, 0x1);
+        dq_reg->DDR_DQ_DQS_RX_SDR_LPDE_CFG[msr][rank] = reg_val;
+
+        // Receiver
+        reg_val = UPDATE_REG_FIELD(dq_reg->DDR_DQ_DQS_RX_IO_CMN_CFG[msr][rank],
+                                    DDR_DQ_DQS_RX_IO_CMN_M1_R0_CFG_CAL_P_C,
+                                    common_cfg->rank[rank].dqs.receiver.code[REC_P_SIDE][REC_C_SIDE]);
 
         reg_val = UPDATE_REG_FIELD(reg_val,
-                                   DDR_DQ_DQS_RX_IO_CMN_M0_R0_CFG_CAL_P_T,
-                                   common_cal->rank[rank_index].dqs.receiver.cal_code[REC_P_SIDE][REC_T_SIDE]);
+                                    DDR_DQ_DQS_RX_IO_CMN_M1_R0_CFG_CAL_P_T,
+                                    common_cfg->rank[rank].dqs.receiver.code[REC_P_SIDE][REC_T_SIDE]);
```

Comment:
alignment of this and below UPDATE_REG_FIELD calls is off

Reply:
Good  catch. Thanks.